### PR TITLE
libaccounts-glib: 1.24 -> 1.26 + some more accounts-sso cleanup

### PR DESCRIPTION
**Pull request by K900**
## Description of changes

https://gitlab.com/accounts-sso/libaccounts-glib/-/commits/1.26

Will be needed for Plasma 6.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by K900** [comment]
Result of `nixpkgs-review pr 278673` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>gnomeExtensions.easyScreenCast</li>
    <li>maui-shell</li>
    <li>xmonad_log_applet</li>
  </ul>
</details>
<details>
  <summary>150 packages built:</summary>
  <ul>
    <li>adapta-gtk-theme</li>
    <li>almanah</li>
    <li>bubblemail</li>
    <li>bubblemail.dist</li>
    <li>calls</li>
    <li>calls.devdoc</li>
    <li>chatty</li>
    <li>cinnamon.cinnamon-common</li>
    <li>cinnamon.cinnamon-screensaver</li>
    <li>digikam</li>
    <li>endeavour</li>
    <li>evolution</li>
    <li>evolution-data-server</li>
    <li>evolution-data-server-gtk4</li>
    <li>evolution-data-server-gtk4.dev</li>
    <li>evolution-data-server.dev</li>
    <li>evolution-ews</li>
    <li>evolutionWithPlugins</li>
    <li>folks</li>
    <li>folks.dev</li>
    <li>folks.devdoc</li>
    <li>gnome-browser-connector</li>
    <li>gnome.geary</li>
    <li>gnome.gnome-applets</li>
    <li>gnome.gnome-calendar</li>
    <li>gnome.gnome-contacts</li>
    <li>gnome.gnome-flashback</li>
    <li>gnome.gnome-notes</li>
    <li>gnome.gnome-panel</li>
    <li>gnome.gnome-panel-with-modules</li>
    <li>gnome.gnome-panel.dev</li>
    <li>gnome.gnome-panel.man</li>
    <li>gnome.gnome-session</li>
    <li>gnome.gnome-session.debug</li>
    <li>gnome.gnome-session.sessions</li>
    <li>gnome.gnome-shell</li>
    <li>gnome.gnome-shell.debug</li>
    <li>gnome.gnome-shell.devdoc</li>
    <li>gnome.gnome-tweaks</li>
    <li>gnome.nixos-gsettings-overrides</li>
    <li>gnomeExtensions.gsconnect</li>
    <li>gnomeExtensions.gsconnect.installedTests</li>
    <li>kile</li>
    <li>kmymoney</li>
    <li>kphotoalbum</li>
    <li>latte-dock</li>
    <li>libaccounts-glib</li>
    <li>libaccounts-glib.dev</li>
    <li>libaccounts-glib.devdoc</li>
    <li>libaccounts-glib.py</li>
    <li>libsForQt5.accounts-qml-module</li>
    <li>libsForQt5.accounts-qt</li>
    <li>libsForQt5.akonadi</li>
    <li>libsForQt5.akonadi-calendar</li>
    <li>libsForQt5.akonadi-calendar-tools</li>
    <li>libsForQt5.akonadi-calendar-tools.dev</li>
    <li>libsForQt5.akonadi-calendar.dev</li>
    <li>libsForQt5.akonadi-contacts</li>
    <li>libsForQt5.akonadi-contacts.dev</li>
    <li>libsForQt5.akonadi-import-wizard</li>
    <li>libsForQt5.akonadi-import-wizard.dev</li>
    <li>libsForQt5.akonadi-mime</li>
    <li>libsForQt5.akonadi-mime.dev</li>
    <li>libsForQt5.akonadi-notes</li>
    <li>libsForQt5.akonadi-notes.dev</li>
    <li>libsForQt5.akonadi-search</li>
    <li>libsForQt5.akonadi-search.dev</li>
    <li>libsForQt5.akonadi.dev</li>
    <li>libsForQt5.akonadiconsole</li>
    <li>libsForQt5.akregator</li>
    <li>libsForQt5.akregator.dev</li>
    <li>libsForQt5.angelfish</li>
    <li>libsForQt5.calendarsupport</li>
    <li>libsForQt5.calendarsupport.dev</li>
    <li>libsForQt5.discover</li>
    <li>libsForQt5.eventviews</li>
    <li>libsForQt5.eventviews.dev</li>
    <li>libsForQt5.falkon</li>
    <li>libsForQt5.gwenview</li>
    <li>libsForQt5.incidenceeditor</li>
    <li>libsForQt5.incidenceeditor.dev</li>
    <li>libsForQt5.kaccounts-integration</li>
    <li>libsForQt5.kaccounts-providers</li>
    <li>libsForQt5.kaddressbook</li>
    <li>libsForQt5.kalarm</li>
    <li>libsForQt5.kamoso</li>
    <li>libsForQt5.kdenlive</li>
    <li>libsForQt5.kdepim-addons</li>
    <li>libsForQt5.kdepim-runtime</li>
    <li>libsForQt5.kdeplasma-addons</li>
    <li>libsForQt5.kdev-php</li>
    <li>libsForQt5.kdev-python</li>
    <li>libsForQt5.kdevelop</li>
    <li>libsForQt5.kdevelop-unwrapped</li>
    <li>libsForQt5.kgpg</li>
    <li>libsForQt5.kio-gdrive</li>
    <li>libsForQt5.kmail</li>
    <li>libsForQt5.kmail-account-wizard</li>
    <li>libsForQt5.kmail.doc</li>
    <li>libsForQt5.kmailtransport</li>
    <li>libsForQt5.kmailtransport.dev</li>
    <li>libsForQt5.knotes</li>
    <li>libsForQt5.koko</li>
    <li>libsForQt5.kontact</li>
    <li>libsForQt5.korganizer</li>
    <li>libsForQt5.kpurpose</li>
    <li>libsForQt5.kpurpose.bin</li>
    <li>libsForQt5.kpurpose.dev</li>
    <li>libsForQt5.libgravatar</li>
    <li>libsForQt5.libgravatar.dev</li>
    <li>libsForQt5.libkdepim</li>
    <li>libsForQt5.libksieve</li>
    <li>libsForQt5.libksieve.dev</li>
    <li>libsForQt5.mailcommon</li>
    <li>libsForQt5.mailcommon.dev</li>
    <li>libsForQt5.mailimporter</li>
    <li>libsForQt5.mauikit-calendar</li>
    <li>libsForQt5.mbox-importer</li>
    <li>libsForQt5.merkuro</li>
    <li>libsForQt5.messagelib</li>
    <li>libsForQt5.messagelib.dev</li>
    <li>libsForQt5.okular</li>
    <li>libsForQt5.pim-data-exporter</li>
    <li>libsForQt5.pim-sieve-editor</li>
    <li>libsForQt5.pimcommon</li>
    <li>libsForQt5.pimcommon.dev</li>
    <li>libsForQt5.plasma-browser-integration</li>
    <li>libsForQt5.plasma-desktop</li>
    <li>libsForQt5.plasma-welcome</li>
    <li>libsForQt5.signond</li>
    <li>libsForQt5.skanpage</li>
    <li>libsForQt5.spectacle</li>
    <li>libsForQt5.tokodon</li>
    <li>libsForQt5.zanshin</li>
    <li>mojave-gtk-theme</li>
    <li>pantheon.elementary-calendar</li>
    <li>pantheon.elementary-greeter</li>
    <li>pantheon.elementary-mail</li>
    <li>pantheon.elementary-session-settings</li>
    <li>pantheon.elementary-tasks</li>
    <li>pantheon.switchboard-plug-onlineaccounts</li>
    <li>pantheon.switchboard-with-plugs</li>
    <li>pantheon.wingpanel-applications-menu</li>
    <li>pantheon.wingpanel-indicator-datetime</li>
    <li>pantheon.wingpanel-with-indicators</li>
    <li>peruse</li>
    <li>planify</li>
    <li>valent</li>
    <li>vimix-gtk-themes</li>
    <li>whitesur-gtk-theme</li>
  </ul>
</details>

**Comment by K900** [comment]
Ping GNOME folks because this does also affect Evolution and others.

File: pkgs/development/libraries/libaccounts-glib/default.nix [hunk 1]
```diff
@@ -3,20 +3,17 @@
...
+  version = "1.26";
 
   outputs = [ "out" "dev" "devdoc" "py" ];
 
   src = fetchFromGitLab {
     owner = "accounts-sso";
     repo = "libaccounts-glib";
     rev = version;
-    sha256 = "0y8smg1rd279lrr9ad8b499i8pbkajmwd4xn41rdh9h93hs9apn7";
+    sha256 = "sha256-KVKylt+XjLfidsS2KzT7oFXP6rTR528lYAUP8dffu7k=";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libaccounts-glib/default.nix:14]:
```suggestion
    hash = "sha256-KVKylt+XjLfidsS2KzT7oFXP6rTR528lYAUP8dffu7k=";
```

**Review by SuperSandro2000 — APPROVED** [review]
LGTM

**Comment by amaxine** [comment]
Just copying what I wrote in matrix - e-d-s shouldn't depend on libaccounts-glib since about 5 years ago[1] and the only remaining reference[2] seems to just be something that was forgotten in cleanup. I'd drop the dependency from e-d-s here or in a separate PR. I'll also open an MR upstream to drop that last mention just for clarity.

[1] https://gitlab.gnome.org/GNOME/evolution-data-server/-/commit/66121edf92f8babe8de8d970faf2155ca8824c07
[2] https://gitlab.gnome.org/GNOME/evolution-data-server/-/blame/3.50.1/CMakeLists.txt#L119

**Pull request merged**